pennylane/_qubit_device.py

```diff
@@ -1631,7 +1631,7 @@ def adjoint_jacobian(
                     f" measurement {m.__class__.__name__}"
                 )
 
-            if m.obs.name == "Hamiltonian":
+            if m.obs.name in ["Hamiltonian", "LinearCombination"]:
```

Comment:
```suggestion
            if not m.obs.has_matrix
```
`LinearCombation` should be supported when it provides a matrix.